js/ui/internetSearch.js

```diff
@@ -54,22 +90,31 @@ const InternetSearchProvider = new Lang.Class({
     },
 
     getResultMetas: function(results, callback) {
-        let metas = results.map(function(resultId) {
+        let metas = results.map(Lang.bind(this, function(resultId) {
             let name;
             if (resultId.startsWith('uri:')) {
                 let uri = resultId.slice('uri:'.length);
                 name = _("Open \"%s\" in browser").format(uri);
             } else if (resultId.startsWith('search:')) {
                 let query = resultId.slice('search:'.length);
-                name = _("Search the web for \"%s\"").format(query);
+                let engineName = this._getEngineName();
+
+                if (engineName) {
+                    /* Translators: the first %s is the search engine name, and the second
+                     * is the search string. For instance, 'Search Google for "hello"'.
+                     */
+                    name = _("Search %s for \"%s\"").format(engineName, query);
+                } else {
+                    name = _("Search the web for \"%s\"").format(query);
```

Comment:
I will change this to say "the internet", but we can't really build the string as you propose. Some languages will need to translate "Search Google for %s" and "Search the internet for %s" as two separate strings, and that's why I did it this way.